backend/app/adapter/db/user_changelog_integration_test.go

```
@@ -0,0 +1,252 @@
+		hasErr                 bool
+	}{
+		{
+			name: "entry for user does not exist",
+			userChangeLogTableRows: []userChangeLogTableRow{
+				{
+					userID:       "12346",
+					email:        "test2@gmail.com",
+					lastViewedAt: monthAgo,
+				},
+			},
+			user: entity.User{
+				ID:    "12345",
+				Name:  "Test User",
+				Email: "test@gmail.com",
+			},
+			expectedLastViewedAt: time.Time{},
+			hasErr:               true,
+		},
+		{
+			name: "user has last viewed time",
+			userChangeLogTableRows: []userChangeLogTableRow{
+				{
+					userID:       "12345",
+					email:        "test@gmail.com",
+					lastViewedAt: monthAgo,
+				},
+			},
+			user: entity.User{
+				ID:    "12345",
+				Name:  "Test User",
+				Email: "test@gmail.com",
+			},
+			expectedLastViewedAt: now,
+			hasErr:               false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			mdtest.AccessTestDB(
+				dbConnector,
+				dbMigrationTool,
+				dbMigrationRoot,
+				dbConfig,
+				func(sqlDB *sql.DB) {
+					insertUserChangeLogTableRows(t, sqlDB, testCase.userChangeLogTableRows)
+
+					userChangeLogRepo := db.NewUserChangeLogSQL(sqlDB)
+
+					_, err := userChangeLogRepo.UpdateLastViewedAt(testCase.user, now)
+					if testCase.hasErr {
+						mdtest.NotEqual(t, nil, err)
+						return
+					}
+
+					mdtest.Equal(t, nil, err)
+
+					lastViewedAt, err := userChangeLogRepo.GetLastViewedAt(testCase.user)
+
+					mdtest.Equal(t, nil, err)
+					mdtest.Equal(t, testCase.expectedLastViewedAt.Unix(), lastViewedAt.Unix())
+				})
+		})
+	}
+}
+
+func TestUserChangeLogSQL_CreateRelation(t *testing.T) {
+	now := time.Now()
+	monthAgo := now.AddDate(0, -1, 0)
+
+	testCases := []struct {
+		name                   string
+		userChangeLogTableRows []userChangeLogTableRow
+		user                   entity.User
+		expectedLastViewedAt   time.Time
+		hasErr                 bool
+	}{
+		{
+			name: "entry for user does not exist",
```

```suggestion
			name: "user does not exist",
```

---

Let's add a case when relation already exists. Should throw an error.